builds/fabric8-che/plugins/keycloak-plugin-server/src/main/java/com/redhat/che/keycloak/server/KeycloakUserChecker.java

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat inc.
+
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat - Initial Contribution
+ *******************************************************************************/
+
+package com.redhat.che.keycloak.server;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.ForbiddenException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.UnauthorizedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+public class KeycloakUserChecker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KeycloakUserChecker.class);
+    private static final String ENDPOINT = "http://che-host:8080/api/token/user";
+
+    /**
+     * Cache that stores mappings from user to authorization status.
+     */
+    private static LoadingCache<String, Boolean> cache = CacheBuilder.newBuilder()
+            .maximumSize(100)
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build(new CacheLoader<String, Boolean>() {
+                @Override
+                public Boolean load(String auth) throws Exception {
+                    try {
+                        String response = new KeycloakHttpJsonRequestFactory().fromUrl(ENDPOINT)
+                                                                              .useGetMethod()
+                                                                              .setAuthorizationHeader(auth)
+                                                                              .request().asString();
+                        return "true".equals(response);
```

Comment:
I don't see why we need to do an exact comparaison. Using parseBoolean would avoid potential issue if we change the implementation of the end point somehow